frontend/dockerfile/dockerfile_test.go

```diff
@@ -7281,3 +7284,151 @@ func fixedWriteCloser(wc io.WriteCloser) filesync.FileOutputFunc {
 		return wc, nil
 	}
 }
+
+// See #4439
+func testExcludedFilesOnCopy(t *testing.T, sb integration.Sandbox) {
```

Comment:
Could we add a couple of more cases (or additional test):
- with a git source (see testAddGit)
- testing that adding a file that does not match does not invalidate cache (see testCopyWildcardCache)

Reply:
Thx for the review. I am traveling at the moment and will resume working on this PR next week. I've moved the PR to draft state until then.